### PR TITLE
lnd_test: fix channel event subscription test flake.

### PR DESCRIPTION
**Pull request by valentinewallace**
In this PR, we fix this Travis flake:

```
--- FAIL: TestLightningNetworkDaemon/multiple_channel_creation_and_update_subscription (13.43s)
        lnd_test.go:75: Failed: (multiple channel creation and update subscription): exited with error: 
            *errors.errorString errored verifying close updates: timeout waiting for channel notifications, only received 3/4 chanupds
            /home/travis/gopath/src/github.com/lightningnetwork/lnd/lnd_test.go:5923 (0xc01d59)
            	testBasicChannelCreationAndUpdates: t.Fatalf("errored verifying close updates: %v", err)
            /home/travis/gopath/src/github.com/lightningnetwork/lnd/lnd_test.go:100 (0xbdfcc6)
            	(*harnessTest).RunTestCase: testCase.test(net, h)
            /home/travis/gopath/src/github.com/lightningnetwork/lnd/lnd_test.go:13401 (0xc3fa84)
            	TestLightningNetworkDaemon.func4: ht.RunTestCase(testCase, lndHarness)
            /home/travis/.gimme/versions/go1.11.5.linux.amd64/src/testing/testing.go:827 (0x4f8caf)
            	tRunner: fn(t)
            /home/travis/.gimme/versions/go1.11.5.linux.amd64/src/runtime/asm_amd64.s:1333 (0x45dd91)
            	goexit: BYTE	$0x90	// NOP
FAIL
```

This flake is caused by a node's channel event subscription request being received after an `inactive channel` event occurs, causing the node to miss a notification.

Flakehunter was run for a long time on this one :)

### CODE REVIEW

**Comment by valentinewallace** [comment]
@halseth PTAL at your convenience :)

File: lnd_test.go [hunk 1]
```diff
@@ -5880,31 +5883,56 @@ func testBasicChannelCreationAndUpdates(net *lntest.NetworkHarness, t *harnessTe
...
 	}
 
-	// Subscribe Alice to channel updates so we can test that both remote
-	// and local force close notifications are received correctly.
-	aliceChanSub := subscribeChannelNotifications(ctxb, t, net.Alice)
-	defer close(aliceChanSub.quit)
+	numChannelUpds = 0
+	for numChannelUpds < totalNtfns {
+		select {
+		case update := <-aliceChanSub.updateChan:
```
Comment (halseth) [inline lnd_test.go]: make this check part of the above loop? Otherwise this looks good to me :)

File: lnd_test.go [hunk 2]
```diff
@@ -5880,31 +5883,56 @@ func testBasicChannelCreationAndUpdates(net *lntest.NetworkHarness, t *harnessTe
...
 	}
 
-	// Since each of the channels just became open, Bob should we receive an
-	// open and an active notification for each channel.
+	// Since each of the channels just became open, Bob and Alice should
+	// each receive an open and an active notification for each channel.
 	var numChannelUpds int
-	for numChannelUpds < 2*numChannels {
+	const totalNtfns = 2 * numChannels
+	for numChannelUpds < totalNtfns {
```
Comment (wpaulino) [inline lnd_test.go]: Could extract this into a closure to avoid repeating the same code twice.

**Comment by Roasbeef** [comment]
Still relevant?

**Comment by halseth** [comment]
cc @valentinewallace

File: lntest/itest/lnd_test.go [hunk 3]
```diff
@@ -5982,31 +5985,47 @@ func testBasicChannelCreationAndUpdates(net *lntest.NetworkHarness, t *harnessTe
...
+			select {
+			case update := <-sub.updateChan:
+				switch update.Type {
+				case lnrpc.ChannelEventUpdate_ACTIVE_CHANNEL:
+					if numChannelUpds%2 != 1 {
+						return fmt.Errorf("expected open" +
+							"channel ntfn, got active " +
+							"channel ntfn instead")
+					}
+				case lnrpc.ChannelEventUpdate_OPEN_CHANNEL:
```
[the remark this answers is not in the document]
Comment (valentinewallace) [reply lntest/itest/lnd_test.go]: fixed, thx :)

File: lnd_test.go [hunk 4]
```diff
@@ -5880,31 +5883,56 @@ func testBasicChannelCreationAndUpdates(net *lntest.NetworkHarness, t *harnessTe
...
+				totalNtfns)
 		}
 	}
 
-	// Subscribe Alice to channel updates so we can test that both remote
-	// and local force close notifications are received correctly.
-	aliceChanSub := subscribeChannelNotifications(ctxb, t, net.Alice)
-	defer close(aliceChanSub.quit)
+	numChannelUpds = 0
+	for numChannelUpds < totalNtfns {
```
[the remark this answers is not in the document]
Comment (cfromknecht) [reply lnd_test.go]: ignore the super old comment lol

**Pull request merged**